src/pages/ProductDetail.jsx

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { useParams } from 'react-router-dom';
+import style from './ProductDetail.module.css';
+import { getProductDetail } from '../data/API';
+import { useState, useEffect } from 'react';
+import Button from '../components/ui/button/Button';
+import { addOrUpdateToCart } from '../api/firebase';
+import { useNavigate } from 'react-router-dom';
+import { useRecoilState } from 'recoil';
+import { loginState, userInfoState } from '../data/LoginData';
+import { BsCartX } from 'react-icons/bs';
+
+const SHIPPING = 3000;
+export default function ProductDetail() {
+  const [isLoggedIn, setIsLoggedIn] = useRecoilState(loginState);
+  const [userInfo, setUserInfo] = useRecoilState(userInfoState);
+  const userName = userInfo.displayName;
+  const { id } = useParams();
+  const navigate = useNavigate();
+  const [detail, setDetail] = useState([]);
+  const { id: productId, title, photo, price, description, isSoldOut } = detail;
+
+  useEffect(() => {
+    const details = getProductDetail(id);
+    details.then((data) => {
+      setDetail(data);
+    });
+  }, [id]);
```

Comment:
```suggestion
  useEffect(() => {
    const details = getProductDetail(id);
    details.then((data) => {
      setDetail(data);
    });
  }, []);
```
일단 상세페이지가 마운트 되면 id 가 바뀔일이 없으니 [] 로 해도 될거 같네요

Reply:
취소~! 번복할게요
물론 id 가 바뀔일 없으니 같은 동작을 하겠지만,  
로직이 id 에 의존하니까 [id] 하는게 더 좋은 코드 같습니다.